sdktests/server_side_persistence_read_write.go

```diff
@@ -0,0 +1,469 @@
+	"github.com/launchdarkly/sdk-test-harness/v2/mockld"
+	"github.com/launchdarkly/sdk-test-harness/v2/servicedef"
+	"github.com/stretchr/testify/require"
+)
+
+func (s *ServerSidePersistentTests) doReadWriteTests(t *ldtest.T) {
+	// No cache is enabled
+	t.Run("initializes store when data received", s.initializesStoreWhenDataReceived)
+	t.Run("applies updates to store", s.appliesUpdatesToStore)
+
+	t.Run("data source updates respect versioning", s.dataSourceUpdatesRespectVersioning)
+	t.Run("data source deletions respect versioning", s.dataSourceDeletesRespectVersioning)
+
+	cacheConfigs := []servicedef.SDKConfigPersistentCache{
+		{Mode: servicedef.Infinite},
+		{Mode: servicedef.TTL, TTL: o.Some(1)},
+	}
+
+	for _, cacheConfig := range cacheConfigs {
+		t.Run(fmt.Sprintf("cache mode %s", cacheConfig.Mode), func(t *ldtest.T) {
+			t.Run("does not cache flag miss", func(t *ldtest.T) {
+				s.doesNotCacheFlagMiss(t, cacheConfig)
+			})
+			t.Run("sdk reflects data source updates even with cache", func(t *ldtest.T) {
+				s.sdkReflectsDataSourceUpdatesEvenWithCache(t, cacheConfig)
+			})
+			t.Run("ignores dropped flags", func(t *ldtest.T) {
+				s.ignoresDroppedFlagsWithForeverCache(t)
+			})
+		})
+	}
+
+	t.Run("infinite cache", func(t *ldtest.T) {
+		t.Run("ignores dropped flags", s.ignoresDroppedFlagsWithForeverCache)
+	})
+
+	t.Run("ttl cache", func(t *ldtest.T) {
+		t.Run("ignores dropped flags", s.ignoresDroppedFlagsWithTTLCache)
+	})
+}
+
+func (s *ServerSidePersistentTests) initializesStoreWhenDataReceived(t *ldtest.T) {
+	require.NoError(t, s.persistentStore.Reset())
+
+	persistence := NewPersistence()
+	persistence.SetStore(servicedef.SDKConfigPersistentStore{
+		Type: s.persistentStore.Type(),
+		DSN:  s.persistentStore.DSN(),
+	})
+	persistence.SetCache(servicedef.SDKConfigPersistentCache{
+		Mode: servicedef.Off,
+	})
+
+	sdkData := s.makeSDKDataWithFlag("flag-key", 1, ldvalue.String("value"))
+	_, configurers := s.setupDataSources(t, sdkData)
+	configurers = append(configurers, persistence)
+
+	_, err := s.persistentStore.ReadField("launchdarkly:$inited")
+	require.Error(t, err) // should not exist
+
+	_ = NewSDKClient(t, s.baseSDKConfigurationPlus(configurers...)...)
+	s.eventuallyRequireDataStoreInit(t, "launchdarkly")
+}
+
+func (s *ServerSidePersistentTests) appliesUpdatesToStore(t *ldtest.T) {
+	require.NoError(t, s.persistentStore.Reset())
+
+	persistence := NewPersistence()
+	persistence.SetStore(servicedef.SDKConfigPersistentStore{
+		Type: s.persistentStore.Type(),
+		DSN:  s.persistentStore.DSN(),
+	})
+	persistence.SetCache(servicedef.SDKConfigPersistentCache{
+		Mode: servicedef.Off,
+	})
+
+	sdkData := s.makeSDKDataWithFlag("flag-key", 1, ldvalue.String("value"))
+	stream, configurers := s.setupDataSources(t, sdkData)
+	configurers = append(configurers, persistence)
+
+	_, err := s.persistentStore.ReadField("launchdarkly:$inited")
+	require.Error(t, err) // should not exist
```

Comment:
Is there any ambiguity that this error could be something else besides `not exist`? Like, the connection to the DB is down? 

Basically wondering if the assert here should check the actual error type, more than existence.